strata.js

```diff
@@ -809,17 +809,81 @@
         $.ajax(fetchCases);
     };
 
-    //Search cases SFDC/Calaveras
-    strata.cases.search = function (onSuccess, onFailure, searchQuery, isSolrSearch) {
+    //Utility wrapper for preparing SOLR query
+    function prepareSolrQuery(caseStatus, caseOwner, caseGroup, searchString, sortField, sortOrder, offset, limit, queryParams) {
+        var solrQueryString = "";
+        var identifier = '';
+        var isObjectNothing = function (object) {
+            if (object === '' || object === undefined || object === null) {
+                return true;
+            } else {
+                return false;
+            }
+        };
+        var concatQueryString = function(param){
+            if(solrQueryString === ""){
+                solrQueryString = param;
+            }else{
+                solrQueryString = solrQueryString.concat(" AND " + param);
+            }
+        };        
+        if (!isObjectNothing(caseStatus)) {
+            identifier = '+case_status:';
+            if (caseStatus.toLowerCase() === 'open') {
+                concatQueryString(identifier + 'Waiting*');
+            } else if (caseStatus.toLowerCase() === 'closed') {
+                concatQueryString(identifier + 'Closed*');
+            } else{
+                concatQueryString(identifier + '*');
+            }
+        }
+        if (!isObjectNothing(caseOwner)) { 
+            identifier = '+case_owner:';
+            concatQueryString(identifier + caseOwner);
+        }
+        if (!isObjectNothing(caseGroup)) {
+            identifier = '+case_folderNumber:';
+            if (caseGroup === 'ungrouped') {
+                concatQueryString('+case_hasGroup:false');
+            } else {
+                concatQueryString(identifier + caseGroup);
+            }
+        }
+        if (!isObjectNothing(searchString)) {
+            identifier = 'allText:';
+            concatQueryString(identifier + searchString);
+        }
+        if (!isObjectNothing(queryParams) && queryParams.length > 0){
+            for (var i = 0; i < queryParams.length; ++i) {
+                concatQueryString(queryParams[i]);
+            }
+        }
+        if (!isObjectNothing(sortField)) {
+            identifier = '&sort=case_';
+            solrQueryString = solrQueryString.concat(identifier + sortField);
+        }
+        if (!isObjectNothing(sortOrder)) {
+            solrQueryString = solrQueryString.concat(" " + sortOrder);
+        }
+        if (!isObjectNothing(offset)) {
+            solrQueryString = solrQueryString.concat("&offset=" + offset);
+        }
+        if (!isObjectNothing(limit)) {
+            solrQueryString = solrQueryString.concat("&limit=" + limit);
+        }
+        return solrQueryString;
+    }
+
+    //Search cases SOLR
+    strata.cases.search = function (onSuccess, onFailure, caseStatus, caseOwner, caseGroup, searchString, sortField, sortOrder, offset, limit, queryParams) {
```

Comment:
I think we need two "queryParams" and they should both be strings.  One that is added the the main query string as your currently do. (addlQueryString)??  The other that is appended to the end of the entire query.(addlQueryParams)??  This way the user can extend both the query string  and add additional query options available in the console.[1](fl, df, etc)  Thoughts?

[1] - http://gss-solr01.web.qa.ext.phx1.redhat.com:8888/solr/#/collection1/query